agenda/agenda_presentation/src/main/java/com/alexafit/agendapresentation/overview/OverviewViewModel.kt

```diff
@@ -1,6 +1,58 @@
 package com.alexafit.agendapresentation.overview
 
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.alexafit.agendapresentation.overview.model.OverviewCalendar
+import com.alexafit.agendapresentation.overview.model.OverviewClickEvents
+import com.alexafit.agendapresentation.overview.model.OverviewState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import java.time.LocalDate
+import javax.inject.Inject
 
-class OverviewViewModel: ViewModel() {
-}
\ No newline at end of file
+@HiltViewModel
+class OverviewViewModel @Inject constructor() : ViewModel() {
+    var overviewState by mutableStateOf(OverviewState())
+        private set
+
+    init {
+        updateSixDayOverview()
+    }
+    fun onClickEvent(clickEvent: OverviewClickEvents) {
+        viewModelScope.launch {
+            when (clickEvent) {
+                is OverviewClickEvents.OnDialogSelection -> {
+                    overviewState = overviewState.copy(
+                        startingDate = clickEvent.dialogDate,
+                        currentMonth = clickEvent.dialogDate.month.name,
+                        chosenDate = clickEvent.dialogDate
+                    )
+                    updateSixDayOverview(startingDate = clickEvent.dialogDate)
+                }
+                is OverviewClickEvents.OnOverviewDateSelected -> {
+                    overviewState = overviewState.copy(
+                        chosenDate = clickEvent.selectedDate
+                    )
+                }
+            }
+        }
+    }
+
+    private fun updateSixDayOverview(startingDate: LocalDate? = null) {
+        val startDate = startingDate ?: overviewState.startingDate
+        val listOfSixConsecutiveDays = mutableListOf<OverviewCalendar>()
```

Comment:
`OverviewCalendar` sounds like you have a list of calendars, but isn't it rather a list of days? What about just calling it `Day`?